src/wallet/export/caravan.rs

```diff
@@ -0,0 +1,1005 @@
+    pub fn descriptor(
+        &self,
+        keychain: KeychainKind,
+    ) -> Result<Descriptor<DescriptorPublicKey>, Error> {
+        let required = self.quorum.required_signers;
+        let network: Network = self.network();
+
+        let result = match self.address_type {
+            CaravanAddressType::P2sh => {
+                let keys: Vec<DescriptorKey<Legacy>> = self.descriptor_keys(keychain)?;
+                descriptor! { sh ( sortedmulti_vec(required, keys) ) }
+            }
+            CaravanAddressType::P2shP2wsh => {
+                let keys: Vec<DescriptorKey<Segwitv0>> = self.descriptor_keys(keychain)?;
+                descriptor! { sh ( wsh ( sortedmulti_vec(required, keys) ) ) }
+            }
+            CaravanAddressType::P2wsh => {
+                let keys: Vec<DescriptorKey<Segwitv0>> = self.descriptor_keys(keychain)?;
+                descriptor! { wsh ( sortedmulti_vec(required, keys) ) }
+            }
+        }
+        .map_err(|e| Error::Descriptor(e));
+
+        match result {
+            Ok((d, _, n)) => {
+                if n.contains(&network) {
+                    Ok(d)
+                } else {
+                    Err(Error::InvalidNetwork {
+                        requested: network,
+                        found: *n.iter().last().expect("network"),
+                    })
+                }
+            }
+            Err(e) => Err(e),
+        }
+    }
+
+    fn descriptor_keys<Ctx: ScriptContext>(
+        &self,
+        keychain: KeychainKind,
+    ) -> Result<Vec<DescriptorKey<Ctx>>, DescriptorError> {
+        let result = self
+            .extended_public_keys
+            .iter()
+            .map(|k| {
+                let fingerprint = k.xfp;
+                let key_path = k.bip32_path.clone();
+                let key_source = fingerprint.zip(key_path);
+                let keychain_index = keychain as u32;
+                let derivation_path = DerivationPath::master().child(ChildNumber::Normal {
+                    index: keychain_index,
+                });
+                k.xpub
+                    .into_descriptor_key(key_source, derivation_path)
+                    .map_err(|e| DescriptorError::Key(e))
+            })
+            .collect();
+        result
+    }
+
+    /// Export BDK wallet configuration as a Caravan configuration
+    pub fn export_wallet<D: BatchDatabase>(
+        wallet: &Wallet<D>,
+        name: String,
+        client_type: String,
+    ) -> Result<Self, Error> {
+        let network = wallet.network;
+        let external_descriptor = wallet.get_descriptor_for_keychain(KeychainKind::External);
+        if wallet.change_descriptor.is_none() {
+            return Err(Error::Generic(
+                "Wallet much have an internal descriptor".to_string(),
+            ));
+        }
+        let internal_descriptor = wallet
+            .change_descriptor
+            .as_ref()
+            .expect("internal descriptor");
```

Comment:
Would it better to use a `match` statement here?

Also there seems to be a typo on line 206: `"much" -> "must"`